onebusaway-android/src/main/java/org/onebusaway/android/io/elements/ObaAgencyElement.java

```diff
@@ -114,14 +114,7 @@ public boolean equals(Object obj) {
             return false;
         }
         ObaAgencyElement other = (ObaAgencyElement) obj;
-        if (id == null) {
-            if (other.id != null) {
-                return false;
-            }
-        } else if (!id.equals(other.id)) {
-            return false;
-        }
-        return true;
+        return id.equals(other.id);
```

Comment:
The earlier code checks if id is null, wouldn't this throw a null pointer exception if id is null?

Reply:
Id can never be null.